lib/modules/manager/pub/artifacts.ts

```diff
@@ -63,9 +52,14 @@ export async function updateArtifacts({
       );
     }
 
+    const oldLazyPubspecLock = lazyParsePubspeckLock(
+      lockFileName,
+      oldLockFileContent
+    );
     const constraint = isFlutter
-      ? config.constraints?.flutter ?? getFlutterConstraint(oldLockFileContent)
-      : config.constraints?.dart ?? getDartConstraint(oldLockFileContent);
+      ? config.constraints?.flutter ??
+        oldLazyPubspecLock.getValue()?.sdks.flutter
+      : config.constraints?.dart ?? oldLazyPubspecLock.getValue()?.sdks.dart;
```

Comment:
It reads a bit hard to me. What about something like this:

```ts
const sdk = isFlutter ? 'flutter' : 'dart';
let constraint = config.constraints?.[sdk];
if (!constraint) {
  // Lazy part is not necessary now
  constraint = ...
}
```

Reply:
sounds good to me